plugins/rest/komand_rest/util/util.py

```diff
@@ -61,6 +62,34 @@ def first(sequence, default=""):
     return next((x for x in sequence if x), default)
 
 
+def convert_dict_body_to_string(dict_object: Dict[str, Any]) -> str:
+    """
+    This method will convert a dict object to a string
+    suitable for sending data in x-www-form-urlencoded format
+    :param dict_object: The dict object to convert
+    :return: A new string properly formatted
+    """
+    output_string = ""
+    for key, value in dict_object.items():
+        output_string += f"{key}={value}&"
+    return output_string[:-1]
+
+
+def convert_body_for_urlencoded(headers: Dict[str, str], body: Dict[str, Any]) -> Union[Dict[str, Any], str]:
+    """
+    This method will encode the body if the headers == x-www-form-urlencoded
+    :param headers: Headers dict to read for conditional
+    :param body: Body dict to convert to string with encoding
+    :return: Body as an encoded string value
+    """
+    for key, value in headers.items():
+        if "content-type" in key.lower():
+            if "x-www-form-urlencoded" in value:
+                body = convert_dict_body_to_string(body)
+                break
```

Comment:
I think we should check for an exact match, and exclude partial matches.
Also, you can combine the if-statements into one. If the first boolean statement fails, it will skip the remainder, bypassing a hypothetical `KeyError`.
```suggestion
        if key.lower() == "content-type" and value.lower() == "x-www-form-urlencoded":
                body = urlencode(body)
                break
```
Remember to add `from urllib.parse import urlencode`